client-linuxapp/src/serviceinfo.rs

```diff
@@ -69,6 +74,49 @@ fn create_user(user: &str) -> Result<()> {
     Ok(())
 }
 
+fn check_password_encryption(s: &str) -> bool {
+    let prefixes = ["$2b$", "$6$", "$5$"];
+
+    for prefix in prefixes.iter() {
+        if s.starts_with(prefix) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+fn create_user_with_password(user: &str, password: &str) -> Result<()> {
+    // Checks if user already present
+    let user_info = passwd::Passwd::from_name(user);
+    if user_info.is_some() {
+        log::info!("User {user} is already present");
+        return Ok(());
+    }
+
+    let mut str_encrypted_pw = password.to_string();
+    log::info!("Checking for password encryption");
+    if check_password_encryption(password) == false {
```

Comment:
It would make the goal of the function more clear if you rename it to `is_password_encrypted`, for instance.

Don't compare results to true or false, is best to: `if !is_password_encrypted(password) {`